contributing.md

```diff
@@ -289,6 +289,66 @@ ansible-playbook --skip-tags tests::cleanup \
     -i testhost, tests/playbooks/tests_802_1x.yml
 ```
 
+### Integration tests with podman
+1. Create `~/.ansible/collections/ansible_collections/containers/podman/` if this
+directory does not exist and `cd` into this directory.
+```bash
+mkdir -p ~/.ansible/collections/ansible_collections/containers/podman/
+cd ~/.ansible/collections/ansible_collections/containers/podman/
+```
+
+2. Clone the collection plugins for Ansible-Podman into the current directory.
+```bash
+git clone https://github.com/containers/ansible-podman-collections.git .
+```
+
+3. Change directory into the `network/tests`, `network` is the Repo you cloned from
+your local fork.
+```bash
+cd ~/network/tests
+```
+
+
+4. Use podman with `-d` to run in the background (daemon). Use `c7` because
+`centos/systemd` is centos7.
+```bash
+podman run --name c7 --rm --privileged -v /sys/fs/cgroup:/sys/fs/cgroup:ro -d registry.centos.org/centos/systemd
+```
+
+5. Use `podman unshare` first to run "podman mount" in root mode, use `-vi` to run
+ansible as inventory in verbose mode, use `-c podman` to use collection plugins. Note,
+the failed test files are:
+- `tests_802_1x_nm.yml` 
+- `tests_802_1x_updated_nm.yml`
+- `tests_bond_initscripts.yml`
+- `tests_bond_nm.yml`
+- `tests_bridge_initscripts.yml`
+- `tests_bridge_nm.yml`
+- `tests_default_nm.yml`
+- `tests_ethernet_nm.yml`
+- `tests_reapply_nm.yml`
+- `tests_states_initscripts.yml`
+- `tests_states_nm.yml`
+- `tests_unit.yml`
```

Comment:
What's the failure for this one?

Reply:
- `tests_reapply_nm.yml`

TASK [linux-system-roles.network : Enable and start NetworkManager] **************************************************************************************************************************************************************************
fatal: [c7]: FAILED! => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result", "changed": false}

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
c7                         : ok=16   changed=1    unreachable=0    failed=1    skipped=2    rescued=0    ignored=0   

- `tests_states_initscripts.yml`

TASK [remove test interface if necessary] ****************************************************************************************************************************************************************************************************
fatal: [c7]: FAILED! => {"changed": true, "cmd": ["ip", "link", "del", "statebr"], "delta": "0:00:00.033001", "end": "2020-09-11 02:14:07.996484", "msg": "non-zero return code", "rc": 1, "start": "2020-09-11 02:14:07.963483", "stderr": "Cannot find device \"statebr\"", "stderr_lines": ["Cannot find device \"statebr\""], "stdout": "", "stdout_lines": []}
...ignoring


TASK [shell] *********************************************************************************************************************************************************************************************************************************
fatal: [c7]: FAILED! => {"changed": true, "cmd": "nmcli con delete statebr\nnmcli con load /etc/sysconfig/network-scripts/ifcfg-statebr\nrm -f /etc/sysconfig/network-scripts/ifcfg-statebr\nip link del statebr\n", "delta": "0:00:00.044075", "end": "2020-09-11 02:14:19.344148", "msg": "non-zero return code", "rc": 1, "start": "2020-09-11 02:14:19.300073", "stderr": "Error: Could not create NMClient object: Could not connect: No such file or directory.\nError: Could not create NMClient object: Could not connect: No such file or directory.\nCannot find device \"statebr\"", "stderr_lines": ["Error: Could not create NMClient object: Could not connect: No such file or directory.", "Error: Could not create NMClient object: Could not connect: No such file or directory.", "Cannot find device \"statebr\""], "stdout": "", "stdout_lines": []}
...ignoring


TASK [remove test interface if necessary] ****************************************************************************************************************************************************************************************************
fatal: [c7]: FAILED! => {"changed": true, "cmd": ["ip", "link", "del", "statebr"], "delta": "0:00:00.032218", "end": "2020-09-11 02:14:20.950660", "msg": "non-zero return code", "rc": 1, "start": "2020-09-11 02:14:20.918442", "stderr": "Cannot find device \"statebr\"", "stderr_lines": ["Cannot find device \"statebr\""], "stdout": "", "stdout_lines": []}
...ignoring


TASK [shell] *********************************************************************************************************************************************************************************************************************************
fatal: [c7]: FAILED! => {"changed": true, "cmd": "nmcli con delete statebr\nnmcli con load /etc/sysconfig/network-scripts/ifcfg-statebr\nrm -f /etc/sysconfig/network-scripts/ifcfg-statebr\nip link del statebr\n", "delta": "0:00:00.042914", "end": "2020-09-11 02:14:33.191797", "msg": "non-zero return code", "rc": 1, "start": "2020-09-11 02:14:33.148883", "stderr": "Error: Could not create NMClient object: Could not connect: No such file or directory.\nError: Could not create NMClient object: Could not connect: No such file or directory.\nCannot find device \"statebr\"", "stderr_lines": ["Error: Could not create NMClient object: Could not connect: No such file or directory.", "Error: Could not create NMClient object: Could not connect: No such file or directory.", "Cannot find device \"statebr\""], "stdout": "", "stdout_lines": []}
...ignoring


PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
c7                         : ok=275  changed=28   unreachable=0    failed=0    skipped=72   rescued=0    ignored=4   



- `tests_states_nm.yml`

TASK [remove test interface if necessary] ****************************************************************************************************************************************************************************************************
fatal: [c7]: FAILED! => {"changed": true, "cmd": ["ip", "link", "del", "statebr"], "delta": "0:00:00.034859", "end": "2020-09-11 02:19:41.932751", "msg": "non-zero return code", "rc": 1, "start": "2020-09-11 02:19:41.897892", "stderr": "Cannot find device \"statebr\"", "stderr_lines": ["Cannot find device \"statebr\""], "stdout": "", "stdout_lines": []}
...ignoring


TASK [linux-system-roles.network : Enable and start NetworkManager] **************************************************************************************************************************************************************************
fatal: [c7]: FAILED! => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result", "changed": false}


TASK [fail] **********************************************************************************************************************************************************************************************************************************
fatal: [c7]: FAILED! => {"changed": false, "msg": "!!!!! Failure in test 'I can create a profile' !!!!!"}


TASK [shell] *********************************************************************************************************************************************************************************************************************************
fatal: [c7]: FAILED! => {"changed": true, "cmd": "nmcli con delete statebr\nnmcli con load /etc/sysconfig/network-scripts/ifcfg-statebr\nrm -f /etc/sysconfig/network-scripts/ifcfg-statebr\nip link del statebr\n", "delta": "0:00:00.045038", "end": "2020-09-11 02:19:50.210473", "msg": "non-zero return code", "rc": 1, "start": "2020-09-11 02:19:50.165435", "stderr": "Error: Could not create NMClient object: Could not connect: No such file or directory.\nError: Could not create NMClient object: Could not connect: No such file or directory.\nCannot find device \"statebr\"", "stderr_lines": ["Error: Could not create NMClient object: Could not connect: No such file or directory.", "Error: Could not create NMClient object: Could not connect: No such file or directory.", "Cannot find device \"statebr\""], "stdout": "", "stdout_lines": []}
...ignoring

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
c7                         : ok=26   changed=3    unreachable=0    failed=1    skipped=2    rescued=1    ignored=2   



- `tests_unit.yml`

TASK [Install dependencies] ******************************************************************************************************************************************************************************************************************
ok: [c7] => (item=NetworkManager-libnm) => {"ansible_loop_var": "item", "changed": false, "item": "NetworkManager-libnm", "msg": "", "rc": 0, "results": ["1:NetworkManager-libnm-1.18.4-3.el7.x86_64 providing NetworkManager-libnm is already installed"]}
failed: [c7] (item=python2-gobject-base) => {"ansible_loop_var": "item", "changed": false, "item": "python2-gobject-base", "msg": "No package matching 'python2-gobject-base' found available, installed or updated", "rc": 126, "results": ["No package matching 'python2-gobject-base' found available, installed or updated"]}
failed: [c7] (item=python3-gobject-base) => {"ansible_loop_var": "item", "changed": false, "item": "python3-gobject-base", "msg": "No package matching 'python3-gobject-base' found available, installed or updated", "rc": 126, "results": ["No package matching 'python3-gobject-base' found available, installed or updated"]}
ok: [c7] => (item=python-gobject-base) => {"ansible_loop_var": "item", "changed": false, "item": "python-gobject-base", "msg": "", "rc": 0, "results": ["python-gobject-base-3.22.0-1.el7_4.1.x86_64 providing python-gobject-base is already installed"]}
ok: [c7] => (item=python2-mock) => {"ansible_loop_var": "item", "changed": false, "item": "python2-mock", "msg": "", "rc": 0, "results": ["python2-mock-1.0.1-10.el7.noarch providing python2-mock is already installed"]}
...ignoring


TASK [Check if python3 is available] *********************************************************************************************************************************************************************************************************
fatal: [c7]: FAILED! => {"changed": false, "cmd": "python3 --version", "msg": "[Errno 2] No such file or directory", "rc": 2}
...ignoring


PLAY RECAP ***********************************************************************************************************************************************************************************************************************************
c7                         : ok=11   changed=2    unreachable=0    failed=0    skipped=3    rescued=0    ignored=2